packages/react-ui/src/TableOfContent/test.js

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import { TableOfContent } from ".";
+import { rootReducer } from "./useTOCReducer";
+
+describe("<TableOfContent />", () => {
+  const observeMock = jest.fn();
+  const disconnectMock = jest.fn();
+  beforeAll(() => {
+    global.IntersectionObserver = () => ({
+      observe: observeMock,
+      disconnect: disconnectMock,
+    });
+  });
+  beforeEach(() => {
+    observeMock.mockClear();
+    disconnectMock.mockClear();
+  });
+
+  it("renders with nav items", () => {
+    expect(observeMock.mock.calls.length).toBe(0);
+    const { container, unmount } = render(
+      <>
+        <TableOfContent ids={["it-works-1", "it-works-2"]} />
+        <h3 id="it-works-1">This should appear inside the table of content</h3>
+        <h2 id="it-works-2" data-short-title="this should appear">
+          This text should not appear inside the table of content, its
+          data-short-title attribute should.
+        </h2>
+      </>
+    );
+    expect(container).toMatchSnapshot();
+    expect(observeMock.mock.calls.length).toBe(2);
+    expect(disconnectMock.mock.calls.length).toBe(0);
+    unmount();
+    expect(disconnectMock.mock.calls.length).toBe(1);
+  });
+  it("renders without nav items", () => {
+    const { container } = render(<TableOfContent ids={["dumb-id"]} />);
+    expect(container).toMatchSnapshot();
+  });
+});
+
+describe("TableOfContent reducer", () => {
+  const initialState = {
+    titles: [
+      {
+        active: false,
+        element: null,
+        id: "ok",
+      },
+    ],
+  };
+  it("reduces properly", () => {
+    let newState = rootReducer(initialState, {
+      type: "setActive",
+      payload: "ok",
+    });
+    expect(newState.titles.length).toBe(1);
+    expect(newState.titles[0].active).toBe(true);
+    const newTitles = [
+      { active: false, element: null, id: "ok-1" },
+      { active: true, element: null, id: "ok-2" },
+      { active: false, element: null, id: "ok-3" },
+    ];
+    newState = rootReducer(initialState, {
+      type: "setTitles",
+      payload: newTitles,
+    });
+    expect(newState.titles).toEqual(newTitles);
```

Comment:
I find it clearer to use separate `it` , one for initialState, one for setActive, one for setTitles